### ISSUE TITLE
HA can’t initialize component

### ISSUE THREAD

**Issue by diplix** [bokub/ha-history-stats#1]
the component throws an error in HA 0.36.1 while trying to initialize:

```
17-01-28 11:21:18 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.history_stats
17-01-28 11:21:18 ERROR (MainThread) [custom_components.sensor.history_stats] Template value cannot be converted to timestamp 
17-01-28 11:21:18 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/homeassistant/helpers/entity_component.py", line 336, in _async_process_entity
    new_entity, self, update_before_add=update_before_add
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/homeassistant/helpers/entity_component.py", line 181, in async_add_entity
    yield from entity.async_update()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/coroutines.py", line 210, in coro
    res = func(*args, **kw)
  File "/Users/ix/.homeassistant/custom_components/sensor/history_stats.py", line 149, in async_update
    self.update_period()
  File "/Users/ix/.homeassistant/custom_components/sensor/history_stats.py", line 202, in update_period
    start = end - duration if start is None else start
TypeError: unsupported operand type(s) for -: 'NoneType' and 'NoneType'

```

this is my configuration:

```
sensor:
  - platform: history_stats
    entity_id: light.flur
    state: 'on'
    start: '{{ as_timestamp(now()) - 7 * 24 * 3600 }}'
    end: '{{ as_timestamp(now()) }}'
    name: Flur Nachtlicht letzte 7 Tage

```

**Comment by bokub** [comment]
Well, that is strange, as `Template value cannot be converted to timestamp ` means that one of the rendered values is not a number. I tried your exact config and couldn't get the error

Could you please go to `http://<your-home-assistant>:8123/dev-template` and copy-paste your template, to check how it is rendered ?

**Comment by diplix** [comment]
{{ as_timestamp(now()) - 7 * 24 * 3600 }} → 1485002691.085546
{{ as_timestamp(now()) }} → 1485607543.445772

when i paste the whole config it returns:

```
- platform: history_stats
    entity_id: light.flur
    state: 'on'
    start: '1485002844.419889'
    end: '1485607644.419973'
    name: Flur Nachtlicht letzte 7 Tage
```

i’m going to try to set the templates without quotes …

**Comment by diplix** [comment]
nope, same error with this:

```
sensor:
  - platform: history_stats
    entity_id: light.flur
    state: 'on'
    start: >
      {{ as_timestamp(now()) - 7 * 24 * 3600 }}
    end: >
      {{ as_timestamp(now()) }}
    name: Flur Nachtlicht letzte 7 Tage
```

```
  - platform: history_stats
    entity_id: light.flur
    state: 'on'
    start: >
      1485003082.646774
    end: >
      1485607882.646853
    name: Flur Nachtlicht letzte 7 Tage
```

**Comment by bokub** [comment]
What is your python version ? 

If you run `float('1485003082.646774')` in a python console, what do you get ?

**Comment by diplix** [comment]
```
ix:~ ix$ python
Python 2.7.10 (default, Jul 30 2016, 19:40:32) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> float('1485003082.646774')
1485003082.646774
>>> exit()
ix:~ ix$ python3
Python 3.6.0 (v3.6.0:41df79263a11, Dec 22 2016, 17:23:13) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> float('1485003082.646774')
1485003082.646774
>>> exit()

```

**Comment by diplix** [comment]
this works, though it is of course pointless:

```
sensor:
  - platform: history_stats
    entity_id: light.flur
    state: 'on'
    start: '1485003082.646774'
    end: '1485607882.646853'
    name: Flur Nachtlicht letzte 7 Tage

```

**Comment by diplix** [comment]
uncommenting lines 74 and 75 in history_stats.py made it work for me. i guess parsing the time aliases is buggy.

**Comment by bokub** [comment]
Oh, I think know what the problem is then, and I have fixed it this morning actually (commit #d527da6)
The `now()` function was systematically transformed to `as_timestamp(now())`, but now it's done only for aliases.

Can you try again with the current version of history_sensor.py ? Thanks

**Comment by diplix** [comment]
great. works flawlessly now.
